dataplane/driver.go

```diff
@@ -279,6 +281,20 @@ func StartDataplaneDriver(configParams *config.Config,
 		intDP := intdataplane.NewIntDataplaneDriver(dpConfig)
 		intDP.Start()
 
+		const healthName = "felix-startup"
+		// Set source-destination-check on AWS EC2 instance.
+		if configParams.AWSSrcDstCheck != string(apiv3.AWSSrcDstCheckOptionDoNothing) {
+			go func(check, healthName string, healthAgg *health.HealthAggregator) {
+				log.Infof("Setting AWS EC2 source-destination-check to %s", check)
+				err := aws.UpdateSrcDstCheck(check)
+				if err != nil {
+					log.WithField("src-dst-check", check).Errorf("Failed to set source-destination-check: %v", err)
+					// set not-ready.
+					healthAggregator.Report(healthName, &health.HealthReport{Live: true, Ready: false})
```

Comment:
Does it make sense to report not-ready if we're not going to retry?

Also, probably makes sense to give this its own health check name; that'll make it easier to debug if it fails.

I think I suggested adding a slow retry here since even errors like permissions could be fixed by the user.  WDYT?

Reply:
Thanks for taking a look Shaun.

> Also, probably makes sense to give this its own health check name; that'll make it easier to debug if it fails.

:+1: 

> I think I suggested adding a slow retry here since even errors like permissions could be fixed by the user. WDYT?
>
>Should we back off to an occasional retry instead of trying to filter retriable/non-retriable errors? I'm thinking a non-retriable error could be something like a permissions error; but the user could fix a permissions error at any time. OTOH, if you have a large cluster with thousands of nodes hitting the API, that could cause throttling.

Actually, this is exactly the concern I had in mind. Should we keep trying indefinitely? A periodic fixed/random try from a node is OK but in case of a large cluster this could be an issue, at a minimum get throttled. This is the reason I am explicitly identifying the _retriable errors_ and capping the errors to a maximum of 3 retries. If we think permission issue is a _retriable error_, I can add that as such. But trying infinitely for any error possible IMO leaves this change more susceptible to being throttled. Idea here is that by marking the status as not-ready, we will let the Admin address the error by looking at the logs. Once the error is addressed, admin can restart Felix to retry setting src-dst-check value.

Please let me know what you think.

> Does it make sense to report not-ready if we're not going to retry?

I am reading this as - if it makes sense to set the status as not-ready if we are going to try in fixed/random interval? - I think this is useful. We can possibly set the status to not-ready for an explicit configuration (AWSSrcDstCheck=Enable). By setting the status we want Admin to take a note of the failure.

Reply:
SGTM